src/libraries/System.Private.CoreLib/src/System/Numerics/VectorMath.cs

```diff
@@ -11,68 +10,131 @@ namespace System.Numerics
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<float> ConditionalSelectBitwise(Vector128<float> selector, Vector128<float> ifTrue, Vector128<float> ifFalse)
         {
-            Debug.Assert(Sse.IsSupported);
-            return Sse.Add(a, Sse.Multiply(Sse.Subtract(b, a), t));
+            // This implementation is based on the DirectX Math Library XMVector4NotEqual method
+            // https://github.com/microsoft/DirectXMath/blob/master/Inc/DirectXMathVector.inl
+
+            if (AdvSimd.IsSupported)
+            {
+                return AdvSimd.BitwiseSelect(selector, ifTrue, ifFalse);
+            }
+            else if (Sse.IsSupported)
+            {
+                return Sse.Or(Sse.And(ifTrue, selector), Sse.AndNot(selector, ifFalse));
+            }
+            else
+            {
+                // Redundant test so we won't prejit remainder of this method on platforms without AdvSimd.
+                throw new PlatformNotSupportedException();
+            }
         }
 
-        public static bool Equal(Vector128<float> vector1, Vector128<float> vector2)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<double> ConditionalSelectBitwise(Vector128<double> selector, Vector128<double> ifTrue, Vector128<double> ifFalse)
         {
-            Debug.Assert(Sse.IsSupported);
-            return Sse.MoveMask(Sse.CompareNotEqual(vector1, vector2)) == 0;
+            // This implementation is based on the DirectX Math Library XMVector4NotEqual method
+            // https://github.com/microsoft/DirectXMath/blob/master/Inc/DirectXMathVector.inl
+
+            if (AdvSimd.IsSupported)
+            {
+                return AdvSimd.BitwiseSelect(selector, ifTrue, ifFalse);
+            }
+            else if (Sse2.IsSupported)
+            {
+                return Sse2.Or(Sse2.And(ifTrue, selector), Sse2.AndNot(selector, ifFalse));
+            }
+            else
+            {
+                // Redundant test so we won't prejit remainder of this method on platforms without AdvSimd.
+                throw new PlatformNotSupportedException();
+            }
         }
 
-        public static bool NotEqual(Vector128<float> vector1, Vector128<float> vector2)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool Equal(Vector128<float> vector1, Vector128<float> vector2)
         {
-            Debug.Assert(Sse.IsSupported);
-            return Sse.MoveMask(Sse.CompareNotEqual(vector1, vector2)) != 0;
+            // This implementation is based on the DirectX Math Library XMVector4Equal method
+            // https://github.com/microsoft/DirectXMath/blob/master/Inc/DirectXMathVector.inl
+
+            if (AdvSimd.Arm64.IsSupported)
+            {
+                Vector128<uint> vResult = AdvSimd.CompareEqual(vector1, vector2).AsUInt32();
```

Comment:
I wonder if the following are faster?
```c#
Vector128<uint> vResult = AdvSimd.CompareEqual(vector1, vector2).AsUInt32();
return (AdvSimd.MinPairwise(vResult, vResult).AsUInt64().ToScalar() == 0xFFFFFFFFFFFFFFFFUL);
```

Reply:
Possibly on ARM64. I copied the existing implementation from the DirectX Math Library, which at least been already profiled/optimized for various scenarios.

Reply:
Agree with @echesakovMSFT.

Reply:
Given the short timeframe to get this in and that the existing code already shows gains, I'm going to wait on doing any additional refactorings/improvements until .NET 6.
The existing code works and is what an already vetted implementation is using.